specs/Qowaiv.Specs/Sustainability/EnergyLabel_specs.cs

```diff
@@ -0,0 +1,521 @@
+
+    [Test]
+    public void from_invalid_as_null_with_TryParse()
+        => EnergyLabel.TryParse("invalid input").Should().BeNull();
+
+    [Test]
+    public void with_TryParse_returns_SVO()
+        => EnergyLabel.TryParse("A++").Should().Be(Svo.EnergyLabel);
+}
+
+public class Has_custom_formatting
+{
+    [TestCase("G")]
+    [TestCase("F")]
+    [TestCase("E")]
+    [TestCase("D")]
+    [TestCase("C")]
+    [TestCase("B")]
+    [TestCase("A")]
+    [TestCase("A+")]
+    [TestCase("A++")]
+    [TestCase("A+++")]
+    [TestCase("A++++")]
+    public void _default(string label)
+        => EnergyLabel.Parse(label).ToString().Should().Be(label);
+
+    [Test]
+    public void with_null_pattern_equal_to_default()
+    {
+        using (TestCultures.En_GB.Scoped())
+        {
+            Svo.EnergyLabel.ToString().Should().Be(Svo.EnergyLabel.ToString(default(string)));
+        }
+    }
+
+    [Test]
+    public void with_string_empty_pattern_equal_to_default()
+    {
+        using (TestCultures.En_GB.Scoped())
+        {
+            Svo.EnergyLabel.ToString().Should().Be(Svo.EnergyLabel.ToString(string.Empty));
+        }
+    }
+
+    [Test]
+    public void default_value_is_represented_as_string_empty()
+        => default(EnergyLabel).ToString().Should().BeEmpty();
+
+    [Test]
+    public void unknown_value_is_represented_as_unknown()
+        => EnergyLabel.Unknown.ToString().Should().Be("?");
+
+    [Test]
+    public void with_empty_format_provider()
+    {
+        using (TestCultures.Es_EC.Scoped())
+        {
+            Svo.EnergyLabel.ToString(FormatProvider.Empty).Should().Be("A++");
+        }
+    }
+
+    [Test]
+    public void custom_format_provider_is_applied()
+    {
+        var formatted = Svo.EnergyLabel.ToString("SomeFormat", FormatProvider.CustomFormatter);
+        Assert.AreEqual("Unit Test Formatter, value: 'A++', format: 'SomeFormat'", formatted);
+    }
+
+    [TestCase(null, "A++", "A++")]
+    [TestCase("U", "B", "B")]
+    [TestCase("l", "G", "g")]
+    [TestCase("l", "A++", "a++")]
+    [TestCase("L", "A++", "a++")]
+    public void fomrat_dependent(string format, EnergyLabel svo, string formatted)
+        => svo.ToString(format).Should().Be(formatted);
+}
+
+public class Is_comparable
+{
+    [Test]
+    public void to_null() => Svo.EnergyLabel.CompareTo(null).Should().Be(1);
+
+    [Test]
+    public void to_EnergyLabel_as_object()
+    {
+        object obj = Svo.EnergyLabel;
+        Svo.EnergyLabel.CompareTo(obj).Should().Be(0);
+    }
+
+    [Test]
+    public void to_EnergyLabel_only()
+        => Assert.Throws<ArgumentException>(() => Svo.EnergyLabel.CompareTo(new object()));
+
+    [Test]
+    public void can_be_sorted_using_compare()
+    {
+        var sorted = new[]
+        {
+            default,
+            default,
+            EnergyLabel.F,
+            EnergyLabel.C,
+            EnergyLabel.A(),
+            EnergyLabel.Unknown,
```

Comment:
Unknown is always highest? I would have expected it after the default

Reply:
default/Empty comes direct before G, so that would be strange.